tests/testthat/test-qgis-result.R

```diff
@@ -35,16 +34,137 @@ test_that("qgis_result_*() functions work", {
   expect_identical(qgis_result_status(result), 0L)
   expect_type(qgis_result_stderr(result), "character")
   expect_type(qgis_result_stdout(result), "character")
-  expect_error(qgis_result_single(result, "numeric"), "zero outputs of type")
+  expect_error(qgis_extract_output_by_class(result, "numeric"), "zero outputs of type")
+
+  result$.processx_result$stdout <- ""
+  expect_error(qgis_check_stdout(result), "output could not be captured")
+})
+
+test_that("qgis_extract_output_by_name() functions work", {
+  skip_if_not(has_qgis())
+
+  tmp_gpkg <- qgis_tmp_vector(".gpkg")
+  tmp_gpkg2 <- qgis_tmp_vector(".gpkg")
+
+  result <- qgis_run_algorithm(
+    "native:extractbyattribute",
+    INPUT = system.file("longlake/longlake_depth.gpkg", package = "qgisprocess"),
+    FIELD = "WAYPOINT_I",
+    OPERATOR = 0,
+    VALUE = "10",
+    OUTPUT = tmp_gpkg,
+    FAIL_OUTPUT = tmp_gpkg2,
+    .quiet = TRUE
+  )
+
+  expect_true(all(file.exists(tmp_gpkg, tmp_gpkg2)))
+  qgis_result_clean(result)
+  expect_false(any(file.exists(tmp_gpkg, tmp_gpkg2)))
```

Comment:
These 3 lines are better dropped in each of these new tests, since `qgis_result_clean()` was already tested in the first test.